### ISSUE TITLE
Ampersand in heading breaks diary index links

### ISSUE THREAD

**Issue by fancsali** [vimwiki/vimwiki#1409]
Prior to submitting a new issue make sure to complete these steps:

- [X] Include the VimWiki settings from your `.vimrc`
```
let g:vimwiki_list = [{'path': '~/vimwiki/', 'syntax': 'markdown', 'ext': '.md', 'auto_diary_index': 1, 'diary_caption_level': 6, 'auto_tags': 1, 'listsyms': ' .X'}]
let g:vimwiki_global_ext = 0
let g:vimwiki_auto_header = 1
let g:vimwiki_emoji_enable = 0
```

- [X] Include the syntax you are using (default / Markdown / MediaWiki) -- Markdown
- [X] Provide a detailed description of the problem including **steps to reproduce the issue**.
  - Enable diary captions/headings to be show on the diary page: set `'diary_caption_level': 6`, for example
  - Go to today's note: `<Leader>w<Leader>w`
  - Create a heading containing and ampersand: `## Ham & Spam`
  NOTE: Due to `g:vimwiki_auto_header = 1` there'll be a top level ("h1") heading above this with the relevant date)
  - Go back to the index: `<Leader>wi`
  - Observe the link to the heading in question to be broken: `- [Ham __LinkDescription__ Spam](2024-06-05#Ham __LinkUrl__ Spam)`

- [X] Include the output of `:VimwikiShowVersion`.
```
Version: 2024.01.24
Os: Linux
Vim: 901
Branch: HEAD
Revision: fde35bb
Date: 2024-01-24 20:14:13 -0700
```

**Comment by fancsali** [comment]
Adding a bit more context...

_Escaping_ the ampersand in the heading included in `diary/2024-08-22.md`:
```
# Fizz \& Buzz
```

Will produce a nice looking link in `diary/diary.md` as expected: 
```
- [Fizz & Buzz](2024-08-11#Fizz & Buzz)
```

However, then the _backslash_ is somewhat of an _eyesore_.

Based on the above, I suspect, the issue is probably the `&`-symbol being included in a regex replace unescaped (and thus being interpreted when it should not be.

**Comment by brennen** [comment]
> Based on the above, I suspect, the issue is probably the &-symbol being included in a regex replace unescaped (and thus being interpreted when it should not be.

Yeah, seems quite likely. I'm guessing a fix will be straightforward, but then again I've thought that before...

**Comment by fancsali** [comment]
I think, I've got a fix, in fact...

**Comment by fancsali** [comment]
... see #1436!

**Comment by brennen** [comment]
Thanks!  I'm about to go offline (and off-grid) for a couple of weeks, but I'll try to get that reviewed/merged before too long.
